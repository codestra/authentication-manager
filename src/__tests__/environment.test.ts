describe('environment', () => {
  beforeEach(() => {
    jest.resetModules();
    process.env = {};
  });

  it('should throw an error if not all environment variables JWT_SECRET, JWT_EXPIRES and PASSWORD_RESET_EXPIRES have been set', async () => {
    process.env = {
      JWT_SECRET: 'test',
      JWT_EXPIRES: 'test',
      // PASSWORD_RESET_EXPIRES:''
    };

    try {
      var initEnvironment = require('../environment').initEnvironment;
      initEnvironment();
    } catch (e) {
      expect(e.message).toEqual(
        'Please define all the environment variables JWT_SECRET, JWT_EXPIRES and PASSWORD_RESET_EXPIRES',
      );
    }
  });

  it('should not throw an error if any of the required environments have been set', async () => {
    process.env = {
      JWT_SECRET: 'test',
      JWT_EXPIRES: 'test',
      PASSWORD_RESET_EXPIRES: 'test',
    };

    var initEnvironment = require('../environment').initEnvironment;
    var { JWT_SECRET, JWT_EXPIRES, PASSWORD_RESET_EXPIRES } = require('../environment');
    initEnvironment();

    expect(JWT_SECRET).toEqual('test');
    expect(JWT_EXPIRES).toEqual('test');
    expect(PASSWORD_RESET_EXPIRES).toEqual('test');
  });
});
