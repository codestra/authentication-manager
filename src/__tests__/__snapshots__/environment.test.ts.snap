// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`environment should throw an error if not all environment variables JWT_SECRET, JWT_EXPIRES and PASSWORD_RESET_EXPIRES have been set 1`] = `"Please define all the environment variables JWT_SECRET, JWT_EXPIRES and PASSWORD_RESET_EXPIRES"`;
